akka-projection-core/src/main/scala/akka/projection/StatusObserver.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection
+
+trait StatusObserver[-Envelope] {
+
+  /**
+   * Called when a projection is started.
+   */
+  def started(projectionId: ProjectionId): Unit
+
+  /**
+   * Called when a projection is restarted due to failure.
+   *
+   * The `restartCount` starts at 1 for the first restart after failure and is incremented for each restart that
+   * also failed. The counter is reset when at least one envelope has been processed successfully.
```

Comment:
Or maybe until the `StatusObserver` implementation resets it.

Reply:
The intention is to have an overview of current status in admin UI to be able to quickly see if a projection is stuck in retrying or restarting. If the count is high and increasing then it's stuck. Number of failed attempts since it was good is not possible to see if it's not reset.

Sure, we could delegate the whole responsibility of counting and resetting to the StatusObserver implementation.

Reply:
@jroper You wrote in the ticket:

> The count of errors and the last error are only conveniences, they allow getting the current error and count of errors without going to the logs, but that's not the core of feature being requested here.

Do you think it's adding motivated value to have the restart/retry counters as parameters or should it be up to the implementation of the `StatusObserver` to keep track of that?
It's adding some complexity because we have to bend the usage of RestartSource and `retry`.